assets/app/views/_pod-template.html

```diff
@@ -6,19 +6,35 @@
  -->
 <div class="component-block pod-template-container">
   <div class="component pod-template-block" ng-repeat="container in podTemplate.spec.containers">
-    <div class="pod-template">                              
-      <h4>{{container.name}}</h4>
+    <div class="pod-template">
+      <!-- Only show container name if there's more than one. -->
+      <h4 ng-if="podTemplate.spec.containers.length > 1">{{container.name}}</h4>
       <div class="pod-template-image">
         <span class="fa fa-cube" aria-hidden="true"></span>
-        <span>Image: {{container.image | imageStreamName}}
-          <span ng-if="imagesByDockerReference[container.image]"> (<span class="small" title="{{imagesByDockerReference[container.image].metadata.name}}">{{imagesByDockerReference[container.image].metadata.name.substr(0, 10)}}</span>)</span>
+        Image:
+        <span ng-if="imagesByDockerReference[container.image]">
+          <span ng-attr-title="{{container.image}}">{{imagesByDockerReference[container.image].metadata.name}}</span>
+          <span ng-init="buildOutput = (imagesByDockerReference[container.image] | buildForImage : builds).spec.output.to">
+            <span ng-if="buildOutput.kind === 'ImageStreamTag'" class="small">
+              (from {{buildOutput.name.substring(buildOutput.name.indexOf(":"))}})
+            </span>
+          </span>
+        </span>
```

Comment:
@liggitt I struggled to express this cleanly (and below as well). Let me know if you spot improvements.